src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs

```diff
@@ -44,11 +44,11 @@ protected override Task OnOpenAsync(object? parameter = null)
 
         if (IsModal)
         {
-            _uiVisualizerService.ShowDialogAsync(_windowViewModel, OnWindowCompleted);
+            Task.Run(async () => await _uiVisualizerService.ShowDialogAsync(_windowViewModel, OnWindowCompleted));
```

Comment:
Are you sure you want to patch this to another thread? Then the ui visualizerservice needs to dispatch back to UI again

Reply:
i need this thread not blocked by Show function (to correctly implement tool logic)

Reply:
Just know that you are not really running this on a separate thread, just queueing it in the dispatcher.

Reply:
yes, i understand... i think if i'll have problems with that - will run it truly separately. But tools has a really small simulative appliance - 99% there wouldn't be a problem